src/InnerResultIterator.php

```diff
@@ -65,23 +66,29 @@ class InnerResultIterator implements \Iterator, \Countable, \ArrayAccess
      */
     private $logger;
 
+    protected $count = null;
+
     /**
      * @param mixed[] $parameters
      * @param array[] $columnDescriptors
      */
-    public function __construct(string $magicSql, array $parameters, ?int $limit, ?int $offset, array $columnDescriptors, ObjectStorageInterface $objectStorage, ?string $className, TDBMService $tdbmService, MagicQuery $magicQuery, LoggerInterface $logger)
+    public function __construct(?string $magicSql, ?array $parameters, ?int $limit, ?int $offset, ?array $columnDescriptors, ?ObjectStorageInterface $objectStorage, ?string $className, ?TDBMService $tdbmService, ?MagicQuery $magicQuery, ?LoggerInterface $logger)
```

Comment:
Idea: we really need 2 different constructors here. We could put the constructor private and empty, and create 2 static methods to create an instance.

Reply:
done